### ISSUE TITLE
Multi-TPU Flash: performance issue (converting module to CPU hangs during teardown)

### ISSUE THREAD

**Issue by krshrimali** [Lightning-AI/lightning-flash#1427]
Description will be added... soon!

cc: @kaushikb11 @Borda @ethanwharris

**Comment by krshrimali** [comment]
**Update:**

- I checked with a custom model in PL, it was tearing down without any delay on TPU (for both single and multiple devices).
- I wanted to try the same custom model in Flash, so I exported it as a backbone and registered it to `ImageClassifier`. The model had basic conv layers, batch normalization, relu and dropout.

**Outcome:**

- Flash training and validation works wihtout hanging.
- Flash prediction works on a single TPU device.
- Flash prediction currently errors out on multiple TPU devices.

Learning: if a model is huge (number of params), or the dataset is large, it will definitely take fair amount of time on TPU. And by default, the backbone in `ImageClassifier` is `resnet18` - which in terms of number of params is not small. The dataset in the examples was also large. If a `BoringDataModule` is used with a `BoringModel` - it all works fine.

**Next steps:**

- Benchmark it with Vanilla XLA code + PL codes for the same models and datamodules to check who is at the fault, and where can we focus! :)

I'll be closing this issue, as we don't have a concern in Flash yet. This should be an up-stream issue, and we'll have to scope it in the future. :)
